### ISSUE TITLE
Permissions are not working

### ISSUE THREAD

**Issue by codeanticode** [processing/processing-android#225]
[This sketch](https://github.com/codeanticode/processing-android-tutorials/blob/master/location_permissions/gps_example/gps_example.pde) uses GPS location, but even after enabling both the ACCESS_COARSE_LOCATION and ACCESS_FINE_LOCATION permissions from the mode, it crashes with the following error:

```
06-12 12:30:31.265  2171  2171 E AndroidRuntime: java.lang.RuntimeException: Unable to resume activity {processing.test.gps_example/processing.test.gps_example.MainActivity}: java.lang.SecurityException: "network" location provider requires ACCESS_COARSE_LOCATION or ACCESS_FINE_LOCATION permission.
```

VIBRATE permission also crashes a watch face with the following error:

```
06-13 19:11:02.123  3308  3308 E AndroidRuntime: java.lang.SecurityException: Requires VIBRATE permission
```

although it was set in the manifest file.

**Comment by codeanticode** [comment]
The problem with watch faces has been fixed with https://github.com/processing/processing-android/commit/38193cec28aa547e05b7dd424e7df6b7b9c9dd29, however the other errors happen on Android 6.0, and are due to the new [dangerous permissions](https://developer.android.com/guide/topics/security/permissions.html#normal-dangerous) in 6.0, which require the users to accept the permissions [during runtime](Requesting Permissions at Run Time
).

**Comment by codeanticode** [comment]
Fixed in 3.0.1 with https://github.com/processing/processing-android/commit/43b8ef6cfeb5fed1680e50d6639301a3f547c2c4
